src/towncrier/_builder.py

```diff
@@ -102,6 +102,7 @@ def __call__(self, section_directory: str = "") -> str:
 def find_fragments(
     base_directory: str,
     config: Config,
+    only_check_categories: bool = False,
```

Comment:
This flag is hard do understand.

would it be possible to refactor the `find_fragments` to also return the category for each file ?

In this way, we don't need to implement any filtering here.

We can implement the filtering in `check.py`

-------

No need to implement the following sugestion in this PR.

With such an API, in check.py we can have 2 error message:

* no fragmtents at all
* only ignored fragments